pkg/codegen/go/gen.go

```diff
@@ -3110,8 +3110,10 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 		pkg.functions = append(pkg.functions, f)
 
 		name := tokenToName(f.Token)
-		originalName := name
-		if pkg.names.Has(name) {
+
+		if pkg.names.Has(name) ||
```

Comment:
Renaming to kick in not only based on function name conflict, but also on FResult name conflict (relevant for the bug) and FArgs thrown in.